test/highlvl-async-usage.test.js

```diff
@@ -172,6 +172,25 @@ describe('basic usage of high-level async methods', function(){
 
   });//</it should have aggregated some config as expected>
 
+  it('should maintain object references when merging modules together', function(done) {
+
+    loader.aggregate({
+      dirname: path.resolve(__dirname, './fixtures/test-refs/config'),
+      filter: /(.+)\.js$/
+    }, function (err, config){
+      if (err) { return done(err); }
+
+      assert(config.databases.someDb);
```

Comment:
Just a note: Using try/catch around this will result in better test output when this stuff fails.  In this case, it'll probably never cause any actual inaccuracies in test results, because mocha seems to only have trouble with crashing when it's a many-level-deep-nested describe block.  Still, it's a nit worth pointing out when we see it, since it's pretty easy to fix